vendor/engines/secure_rooms/app/controllers/secure_rooms/card_readers_controller.rb

```diff
@@ -0,0 +1,63 @@
+module SecureRooms
+
+  class CardReadersController < ApplicationController
+
+    admin_tab :all
+    customer_tab :password
+
+    layout "two_column"
+
+    before_action :init_current_facility
+    before_action :init_product
+    before_action :load_card_reader, except: [:index]
+    before_action :authenticate_user!
+    before_action :check_acting_as
+
+    def initialize
+      @active_tab = "secure_rooms"
+      super
+    end
+
+    def index
+      @card_readers = @product.card_readers
+    end
+
+    def new; end
+
+    def edit; end
+
+    def create
+      @card_reader.assign_attributes(card_reader_params)
```

Comment:
To be consistent with our other controllers, we would normally write this as
```ruby
if @card_reader.update_attributes(card_reader_params)
  redirect_to facility_secure_room_card_readers_path(current_facility, @product)
else
  render :new
end
```

Same structure applies to `update`